sunkit_image/coalignment.py

```diff
@@ -1,284 +1,55 @@
-        columns). The parameters ``y0, y1, x0, x1`` have the type
-        `~astropy.units.Quantity`.
-    """
-    return (
-        [_lower_clip(y.value), _upper_clip(y.value)] * u.pix,
-        [_lower_clip(x.value), _upper_clip(x.value)] * u.pix,
-    )
-
-
-def _upper_clip(z):
-    """
-    Find smallest integer bigger than all the positive entries in the input
-    array.
-    """
-    zcond = z >= 0
-    return int(np.max(np.ceil(z[zcond]))) if np.any(zcond) else 0
-
-
-def _lower_clip(z):
-    """
-    Find smallest positive integer bigger than the absolute values of the
-    negative entries in the input array.
-    """
-    zcond = z <= 0
-    return int(np.max(np.ceil(-z[zcond]))) if np.any(zcond) else 0
-
-
-def match_template_to_layer(layer, template):
-    """
-    Calculate the correlation array that describes how well the template
-    matches the layer. All inputs are assumed to be numpy arrays.
-
-    Parameters
-    ----------
-    layer : `numpy.ndarray`
-        A numpy array of size ``(ny, nx)``.
-    template : `numpy.ndarray`
-        A numpy array of size ``(N, M)`` where ``N < ny`` and ``M < nx``.
-
-    Returns
-    -------
-    `numpy.ndarray`
-        A correlation array between the layer and the template.
-        The values in the array range between 0 and 1.
-    """
-    return match_template(layer, template)
+__all__ = ["coalignment_interface", "register_coalignment_method", "registered_methods"]
 
+## This will be further replaced, once the decorator structure is in place.
+registered_methods = {}
 
-def _find_best_match_location(corr):
-    """
-    Calculate an estimate of the location of the peak of the correlation result
-    in image pixels.
-
-    Parameters
-    ----------
-    corr : `numpy.ndarray`
-        A 2D correlation array.
-
-    Returns
-    -------
-    `~astropy.units.Quantity`
-        The shift amounts ``(y, x)`` in image pixels. Subpixel values are
-        possible.
-    """
-    # Get the index of the maximum in the correlation function
-    ij = np.unravel_index(np.argmax(corr), corr.shape)
-    cor_max_x, cor_max_y = ij[::-1]
-
-    # Get the correlation function around the maximum
-    array_maximum = corr[
-        np.max([0, cor_max_y - 1]) : np.min([cor_max_y + 2, corr.shape[0] - 1]),
-        np.max([0, cor_max_x - 1]) : np.min([cor_max_x + 2, corr.shape[1] - 1]),
-    ]
-    y_shift_maximum, x_shift_maximum = _get_correlation_shifts(array_maximum)
 
-    # Get shift relative to correlation array
-    y_shift_correlation_array = y_shift_maximum + cor_max_y * u.pix
-    x_shift_correlation_array = x_shift_maximum + cor_max_x * u.pix
+def register_coalignment_method(name, method):
+    registered_methods[name] = method
 
-    return y_shift_correlation_array, x_shift_correlation_array
 
+############################ Coalignment Interface begins #################################
+def replace_nan_with_mean(array):
```

Comment:
I assume the scikit image code can't handle nans?

Reply:
> I assume the scikit image code can't handle nans?

Yeah, it does expect that the arrays contain finite values.

> Also why mean instead of 0?

I've included the mean calculation simply to make things easier for the user. This way, they don't have to check if all values in their map are finite. This(mean) is particularly beneficial for HMI maps with NaN values, where the mean provides a more accurate data representation.

Reply:
I think this will need to be user pickable. The rotate method has a missing keyword that lets you pick the value.

Reply:
> I think this will need to be user pickable. The rotate method has a missing keyword that lets you pick the value.

Yeah, agreed. I'll add that too.